src/nitrokey/nk3/updates.py

```diff
@@ -119,9 +119,8 @@ def confirm_extra_information(self, extra_info: List[str]) -> None:
     def confirm_update_same_version(self, version: Version) -> None:
         pass
 
-    @abstractmethod
     def request_repeated_update(self) -> Optional[Exception]:
-        pass
+        raise DeprecationWarning
```

Comment:
While `DeprecationWarning` is indeed an exception, it should be triggered with [`warnings.warn`](https://docs.python.org/3/library/warnings.html#warnings.warn):

```suggestion
        warnings.warn("UpdateUi.request_repeated_update is no longer needed", DeprecationWarning)
        return None
```